### PR TITLE
Feature/24 new terms n workflow implementation

### PR DESCRIPTION
**Pull request by wayangalihpratama**
TODO / DONE

---

- #24 
- [x] Modify all migrations, models, and crud based on updated LLD (refactor table name, add new column, etc)
- [x] Modify the current endpoint based on the above changes
- [x] Implement new migration and model (Business Unit and User Business Unit)
- [x] Modify user API based on updated LLD (adding business unit detail)
  - For now, the business_unit_detail already added to some user response, but the value is None.
  - Waiting the clarity of how we add user business unit

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205713797776421

### CODE REVIEW

File: backend/tests/test_020_case.py [hunk 1]
```diff
@@ -6,64 +6,64 @@
...
             headers={"Authorization": f"Bearer {admin_account.token}"},
         )
         assert res.status_code == 404
 
     @pytest.mark.asyncio
-    async def test_create_project(
+    async def test_create_case(
         self, app: FastAPI, session: Session, client: AsyncClient
     ) -> None:
         payload = {
```
Comment (dedenbangkit) [inline backend/tests/test_020_case.py:35]: also try with private: true

File: backend/models/case.py [hunk 2]
```diff
@@ -19,75 +19,76 @@ class LivingIncomeStudyEnum(enum.Enum):
...
-    multiple_crops: bool
+    multiple_commodities: bool
     logo: Optional[str]
     created_by: int
-    project_crops: List[SimplifiedProjectCropDict]
+    case_commodities: List[SimplifiedCaseCommodityDict]
 
 
-class ProjectDetailDict(TypedDict):
+class CaseDetailDict(TypedDict):
```
Comment (dedenbangkit) [inline backend/models/case.py:52]: private?

File: backend/models/case.py [hunk 3]
```diff
@@ -135,129 +136,132 @@ def __init__(
...
+        self.private = private
         self.created_by = created_by
 
     def __repr__(self) -> int:
-        return f"<Project {self.id}>"
+        return f"<Case {self.id}>"
 
     @property
-    def serialize(self) -> ProjectDict:
+    def serialize(self) -> CaseDict:
```
Comment (dedenbangkit) [inline backend/models/case.py:175]: private

File: backend/models/case.py [hunk 4]
```diff
@@ -135,129 +136,132 @@ def __init__(
...
-            "diversified_crops_count": len(diversified_count),
+            "focus_commodity": self.focus_commodity,
+            "diversified_commodities_count": len(diversified_count),
             "created_at": self.created_at.strftime('%Y-%m-%d %H:%M:%S'),
             "created_by": self.created_by_user.email,
         }
 
     @property
-    def to_project_detail(self) -> ProjectDetailDict:
+    def to_case_detail(self) -> CaseDetailDict:
```
Comment (dedenbangkit) [inline backend/models/case.py:214]: private

File: backend/tests/test_002_user.py [hunk 5]
```diff
@@ -272,12 +276,13 @@ async def test_update_user_password(
...
-    async def test_invite_user_with_by_admin(
+    async def test_invite_user_by_super_admin(
         self, app: FastAPI, session: Session, client: AsyncClient
     ) -> None:
         user_payload = {
-            "fullname": "Invited User",
-            "email": "invited_user@akvo.org",
+            "fullname": "Invited Admin",
+            "email": "admin@akvo.org",
+            "role": UserRole.admin.value,
```
Comment (dedenbangkit) [inline backend/tests/test_002_user.py:285]: business unit is required when it's admin, editor, or viewer

**Review by dedenbangkit — CHANGES_REQUESTED** [review]

File: backend/tests/test_002_user.py (hunk 5 above)
Comment (wayangalihpratama) [reply backend/tests/test_002_user.py:473]: I have pushed an update for this

File: backend/tests/test_002_user.py
[the remark this answers is not in the document]
Comment (wayangalihpratama) [reply backend/tests/test_002_user.py:1]: related to below updates, the user test with business unit also updated.

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**